website-custom-views/src/content/api-reference/commercetools-frontend-application-shell.mdx

```diff
@@ -0,0 +1,499 @@
+---
+title: "@/application-shell"
+---
+
+<Subtitle>
+
+React component that ties together all the main functionalities of the Custom Views.
+
+</Subtitle>
+
+# Installation
+
+```bash
+yarn add @commercetools-frontend/application-shell
+
+# or
+
+npm --save install @commercetools-frontend/application-shell
+```
+
+Additionally install the peer dependencies (if not present)
+
+```bash
+yarn add @apollo/client react react-dom react-intl react-redux react-router-dom redux @testing-library/react @testing-library/react-hooks
+
+# or
+
+npm --save install @apollo/client react react-dom react-intl react-redux react-router-dom redux @testing-library/react @testing-library/react-hooks
+```
+
+# Components
+
+## CustomViewShell
+
+This is the main component that contains all the general logic to render a Custom View.<br />
+The `CustomViewShell` component is expected to be rendered as the top-level component of your application.
+
+### Usage
+
+We recommend to render the application content as `children` of `<CustomViewShell>`.
+
+This allows the `<CustomViewShell>` to pre-configure the application entry point routes. In addition to that, the entry point route is protected by the basic `View` permission check. This means that a user without permissions of your Custom View won't be able to access the Custom View route.
+
+Bear in mind you're not required to use routes in your Custom View. Use them only if you need to have different views.
```

Comment:
Hi @emmenko, could you share in which scenario one would need routes in Custom Views?

I might be lacking info, but I understood that Custom Views will be part of a Native Application's page.

Reply:
Some examples of using routes:
* A list page with details pages
* Tabular pages

> I understood that Custom Views will be part of a Native Application's page

Yes but routes are still allowed. We can double check but routes within a Custom View should not affect the routes of the app, as it's rendered within an iframe.